syntax = "proto3";

package PbTelemetry;

enum TelemetryState {
    INIT = 0;
    IDLE = 1;
    SETUP_RUN = 2;
    RUN = 3;
    WAIT_READY = 4;
    ERROR = 5;
}

message MessagesPerSecond
{
    string deviceName = 1;
    uint64 bitsPerSecond = 2;
    double busLoad = 3;
    uint64 count = 4;
}

message Status
{
    uint64 timestamp = 1;
    uint64 zeroTimestamp = 2;
    TelemetryState state = 3;
    uint64 cpuTotalLoad = 4;
    uint64 cpuProcessLoad = 5;
    uint64 memProcessLoad = 6;
    uint64 canlibBuildTime = 7;
    uint64 telemetryBuildTime = 8;
    repeated MessagesPerSecond messagesPerSecond = 10;
}